packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -857,6 +862,67 @@ export class GarbageCollector implements IGarbageCollector {
 		return this.summaryStateTracker.refreshLatestSummary(result);
 	}
 
+	/**
+	 * Process a GC message.
+	 * @param message - The GC message from the container runtime.
+	 * @param local - Whether it was send by this client.
+	 */
+	public processMessage(message: ISequencedDocumentMessage, local: boolean) {
+		const gcMessageContents = message.contents as IGarbageCollectionMessage;
+		switch (gcMessageContents.type) {
+			case GarbageCollectionMessageType.Sweep: {
+				// Delete the nodes whose ids are present in the contents.
+				const sweepMessageContents = gcMessageContents.contents;
+				this.deleteSweepReadyNodes(sweepMessageContents.deletedNodeIds);
+				break;
+			}
+			default: {
+				break;
+			}
+		}
+	}
+
+	/**
+	 * Delete nodes that are sweep ready. Call the runtime to delete these nodes and clear
+	 * the unreferenced state tracking for them.
+	 * @param sweepReadyNodeIds - The ids of nodes that are ready to be deleted.
+	 */
+	private deleteSweepReadyNodes(sweepReadyNodeIds: string[]) {
+		// The ids in the sweep ready nodes do not contain DDS node ids. This is an optimization to reduce the size
+		// of the GC op. Since GC applies to data store only, all its DDSes are deleted along with it. So, get the
+		// DDS nodes ID from the unreferenced nodes state.
+		const allSweepReadyNodeIds = Array.from(sweepReadyNodeIds);
+		// Use a set for lookup because its much faster than array or map.
+		const sweepReadyNodesSet: Set<string> = new Set(sweepReadyNodeIds);
+		for (const [id] of this.unreferencedNodesState) {
+			// Ignore data store nodes.
+			const pathParts = id.split("/");
+			if (pathParts.length <= 2) {
+				continue;
+			}
+
+			// Get the data store id part. Note that this may include blobs but that's okay since the part would just
+			// be "_blobs" and it won't be found.
+			const dsId = `/${pathParts[1]}`;
+			if (sweepReadyNodesSet.has(dsId)) {
+				allSweepReadyNodeIds.push(id);
+			}
+		}
+		const deletedNodeIds = this.runtime.deleteSweepReadyNodes(allSweepReadyNodeIds);
+
```

Comment:
redundate `Array.from` since it's already a list

Reply:
New topic: Is it possible different clients would end up deleting different sets of nodes, given the same list in the op?  I've never really grasped the case where `this.runtime.deleteSweepReadyNodes` wouldn't delete everything.

Reply:
I used `Array.from` to not change the incoming list.

Reply:
> New topic: Is it possible different clients would end up deleting different sets of nodes, given the same list in the op? I've never really grasped the case where this.runtime.deleteSweepReadyNodes wouldn't delete everything.

They should not. Ideally, runtime would delete everything GC asked for. However, it may choose not to. One example is that we have separate configs for disabling sweep for data stores vs blobs. It's possible that one or both of those settings are on.
Basically, giving us fliexibility to do something like the above where the runtime is in final control of what to delete and GC's job is to only relay information to the runtime but not actually delete naything - that power lies with the runtime.

Reply:
Let's see... If Client A has Blob Sweep disabled but Client B doesn't... If Client A sweeps, then the Blobs will remain, but they'll still be SweepReady, and if Client B ever sweeps, they'll be deleted (included in another GC Sweep op and all).  Seems fine.

Reply:
(feel free to resolve)